### ISSUE TITLE
Feature Request: Support Client Generated Dynamic field searching/terms

### ISSUE THREAD

**Issue by awheeler** [rashidkpc/kibana2#121]
Turns out that ElasticSearch allows you to use the output of a script to do searching/terms.  Here's an example:
https://gist.github.com/3974184

This queries out the  value of pin= from a field named params, and does a terms on it.

So create an interface to allow users to define their own, after-the-fact, searching of fields embedded within the data but not pulled into separate fields.

**Comment by dav3860** [comment]
Yes that would be a great feature. Ideally using external scripts and/or a user-defined query in the search bar.
